python/pyspark/sql/pandas/serializers.py

```diff
@@ -505,7 +510,12 @@ def _create_batch(self, series):
 
         arrs = []
         for s, t in series:
-            if self._struct_in_pandas == "dict" and t is not None and pa.types.is_struct(t):
+            if (
+                self._struct_in_pandas == "dict"
+                and t is not None
+                and pa.types.is_struct(t)
+                and not is_variant(t)
+            ):
                 # A pandas UDF should return pd.DataFrame when the return type is a struct type.
```

Comment:
Why do we want to avoid this path for variant types? Maybe we should leave a comment explaining it.